pkg/graveler/graveler.go

```diff
@@ -2777,6 +2777,10 @@ func (g *Graveler) Merge(ctx context.Context, repository *RepositoryRecord, dest
 		}
 		metadata[MergeStrategyMetadataKey] = mergeStrategyString[mergeStrategy]
 		commit.Metadata = metadata
+		commitID, err = g.RefManager.AddCommit(ctx, repository, commit)
```

Comment:
I don't understand where `CommitID` is being passed to the hook..

Reply:
I think you'd need to put it on commit for this to work. This may demonstrate that this portion of the code needs a unit test.

Reply:
Should just modify TestGraveler_PreMergeHook accordingly

Reply:
@ozkatz that's the only blocking comment from my end - it seems like we're not passing the new `commitID` to the pre merge hook, which was the sole purpose of this change, no?